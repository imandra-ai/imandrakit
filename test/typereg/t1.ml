open Util_

type foo = int [@@deriving typereg]
type bar = foo * int list option [@@deriving typereg]

module Yolo = struct
  type t = float option list array
  [@@deriving typereg] [@@typereg.name "Yolo.t"]
end

let () =
  print_endline "t1:";
  dump ()
