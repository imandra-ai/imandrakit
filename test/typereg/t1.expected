t1:
[{ path = "Dune__exe__T1"; name = "bar"; params = [];
   decl =
   (Alias
      (Tuple
         [(Cstor ("foo", []));
           (Cstor ("option", [(Cstor ("list", [(Cstor ("int", []))]))]))]))
   }]
[{ path = "Dune__exe__T1"; name = "foo"; params = [];
   decl = (Alias (Cstor ("int", []))) }]
[{ path = "Dune__exe__T1"; name = "t"; params = [];
   decl =
   (Alias
      (Cstor ("array",
         [(Cstor ("list", [(Cstor ("option", [(Cstor ("float", []))]))]))])))
   }]
